core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalDateHistogram.java

```diff
@@ -396,6 +396,12 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
         } else {
             // sorted by sub-aggregation, need to fall back to a costly n*log(n) sort
             CollectionUtil.introSort(reducedBuckets, order.comparator());
+            if (reducedBuckets.size() == 1) {
+                // hack: force check of sub-aggregation names and fields if there is only 1 bucket (sort code bypassed)
```

Comment:
Currently if the (date) histogram is ordered by sub-aggregation(s), the order path is validated during the reduce phase. This check occurs implicitly in the [comparator](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/search/aggregations/bucket/MultiBucketsAggregation.java#L67) when the histogram buckets are sorted. However if there are 0 or 1 buckets, there is nothing to sort so this code was bypassed. I added a hack here to catch the case with 1 bucket. To catch the case with 0 buckets, the validation code needs to be refactored to run during the query parsing phase (if possible). This would require parsing all sub-aggregations first and then validating the order path.